lib/client/api.go

```diff
@@ -796,8 +796,43 @@ func WithMakeCurrentProfile(makeCurrentProfile bool) RetryWithReloginOption {
 	}
 }
 
+// NewNonRetryableError wraps an error to indicate that the error should fail
+// IsErrorResolvableWithRelogin. This wrapper is used to workaround the false
+// positives like trace.IsBadParameter check in IsErrorResolvableWithRelogin.
+func NewNonRetryableError(err error) error {
+	return &nonRetryableError{
+		Err: err,
+	}
+}
+
+type nonRetryableError struct {
+	Err error
+}
+
+func (e *nonRetryableError) Error() string {
+	if e == nil || e.Err == nil {
+		return "error"
```

Comment:
I wonder if this should be just `return ""` - meaning that a nil error is no error. WDYT?

Reply:
I changed to empty string. otherwise maybe we can do `return "nil error"` but i like `""` better.